Source/WebCore/Modules/modern-media-controls/controls/abstract-slider.js

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (C) 2016, 2022 Apple Inc. All Rights Reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+class AbstractSlider extends LayoutNode
+{
+    constructor(layoutDelegate, cssClassName = "")
+    {
+        super(`<div class="slider ${cssClassName}"></div>`);
+
+        this._layoutDelegate = layoutDelegate;
+
+        this.appearanceContainer = new LayoutNode(`<div class="slider-appearance"></div>`);
```

Comment:
So is this basically the same thing as the old `<div class="custom-slider"></div>`?

NIT: Maybe we just do `<div class="appearance"></div>` since we're already inside a `<div class="slider"></div>` (i.e. the `slider-` is kinda redundant)?

Reply:
Yes, it was a renaming for something that didn't feel quite obvious. I'll drop the `slider-` prefix!